forward/events.go

```diff
@@ -0,0 +1,64 @@
+package forward
+
+// An EventListener handles events given to it by the SWIM node. HandleEvent should be thread safe.
+type EventListener interface {
+	HandleEvent(interface{})
```

Comment:
Can we refrain from polluting the exported API of a type by making an event listener a `func (interface{})` instead?